src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -343,6 +357,48 @@ private void validateOperationProperties(@Nullable Schema schema, FailureCollect
           .withConfigElement(NAME_TABLE_KEY, keyField);
       }
     }
+
+    Map<String, Integer> keyMap = new HashMap<>();
+    keyFields.forEach(key -> {
+      Integer count = keyMap.get(key);
+      keyMap.put(key, count == null ? 1 : count + 1);
+    });
+
+    keyMap.keySet().stream()
+      .filter(key -> keyMap.get(key) != 1)
+      .forEach(key -> collector.addFailure(
+        String.format("Table key field '%s' is duplicated.", key),
+        String.format("Remove duplicates of Table key field '%s'.", key))
+        .withConfigElement(NAME_TABLE_KEY, key)
+      );
+
+    if (Operation.UPSERT.equals(getOperation()) && getOrderedBy() != null) {
+      List<String> orderedByFields = Arrays.stream(Objects.requireNonNull(getOrderedBy()).split(","))
+        .map(String::trim).collect(Collectors.toList());
+      orderedByFields.stream()
+        .filter(orderedByField -> !fields.contains(orderedByField.split(":")[0]))
```

Comment:
can we have additional class with explicit field names to avoid references by index? It's not clear what it is and what is under index 1 for example